src/mjolnir/ingest_transit.cc

```diff
@@ -280,101 +369,127 @@ bool write_stop_pairs(Transit& tile,
         // add information from calendar_dates.txt
-        gtfs::CalendarDates trip_calDates = feed.get_calendar_dates(currTrip.service_id);
+        const gtfs::CalendarDates& trip_calDates = feed.get_calendar_dates(currTrip->service_id);
         for (const auto& cal_date_item : trip_calDates) {
-
           auto d = (cal_date_item.date).get_raw_date();
           if (cal_date_item.exception_type == gtfs::CalendarDateException::Added)
             stop_pair->add_service_added_dates(stoi(d));
           else
             stop_pair->add_service_except_dates(stoi(d));
         }
-        gtfs::CalendarItem trip_calendar = *(feed.get_calendar(currTrip.service_id));
 
-        stop_pair->set_service_start_date(stoi(trip_calendar.start_date.get_raw_date()));
-        stop_pair->set_service_end_date(stoi(trip_calendar.end_date.get_raw_date()));
-        stop_pair->set_trip_headsign(currTrip.trip_headsign);
+        // check that a valid calendar exists for the service id
+        auto trip_calendar = feed.get_calendar(currTrip->service_id);
+        if (trip_calendar) {
+          stop_pair->set_service_start_date(stoi(trip_calendar->start_date.get_raw_date()));
+          stop_pair->set_service_end_date(stoi(trip_calendar->end_date.get_raw_date()));
+        }
+
+        // grab the headsign
+        stop_pair->set_trip_headsign(currTrip->trip_headsign);
 
         uniques.lock.lock();
-        uniques.trips.insert({currTrip.trip_id, uniques.trips.size()});
-        stop_pair->set_trip_id(stoi(currTrip.trip_id));
+        auto inserted = uniques.trips.insert({currTrip->trip_id, uniques.trips.size()});
+        stop_pair->set_trip_id(inserted.first->second);
         uniques.lock.unlock();
 
-        stop_pair->set_wheelchair_accessible(currTrip.wheelchair_accessible == gtfs::TripAccess::Yes);
+        stop_pair->set_wheelchair_accessible(currTrip->wheelchair_accessible ==
+                                             gtfs::TripAccess::Yes);
 
         // get frequency info
-        auto currFrequencies = feed.get_frequencies(currTrip.trip_id);
-        auto freq_start_time = (currFrequencies[0].start_time.get_raw_time());
-        auto freq_end_time = (currFrequencies[0].start_time.get_raw_time());
-        auto freq_time = freq_start_time + freq_end_time;
-        // TODO: Determine what to do if there are multiple frequencies.
-        // currently, choose fist one
-        if (currFrequencies.size() > 0) {
-          stop_pair->set_frequency_end_time(DateTime::seconds_from_midnight(freq_end_time));
-          stop_pair->set_frequency_headway_seconds(currFrequencies[0].headway_secs);
-        }
+        if (!feed.get_frequencies(currTrip->trip_id).empty()) {
+          const auto& currFrequencies = feed.get_frequencies(currTrip->trip_id);
+          auto freq_start_time = (currFrequencies[0].start_time.get_raw_time());
+          auto freq_end_time = (currFrequencies[0].start_time.get_raw_time());
```

Comment:
i know this didnt really change but this looks wrong. its using the same value as the start time